pkg/controller/controller.go

```diff
@@ -522,3 +523,38 @@ func isInstanceReady(instance *v1alpha1.Instance) bool {
 
 	return false
 }
+
+func IsOwnedBy(obj metav1.Object, owner metav1.Object) bool {
+	ref := GetControllerOf(obj)
+	if ref == nil {
+		return false
+	}
+	return ref.UID == owner.GetUID()
```

Comment:
if there no concrete example of such a bug, I would prefer to check only necessary fields, and UID should be enough. Also see the example from Kubernetes codebase: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/garbagecollector/garbagecollector.go#L231